### ISSUE TITLE
consecutive parallel_apply call accumulates memory usage

### ISSUE THREAD

**Issue by kkawabat** [nalepae/pandarallel#56]
I have a script that uses pandarallel's parallel_apply() multiple times. Looking at memory profiler it seems that pandarallel accumulates in memory after every parallel_apply() call. 

below plot is generated with memory_profiler
```
mprof run --include-children --multiprocess migraine_diary.py 
mprof plot
```
![with_parallel_apply_wo_reinitialization](https://user-images.githubusercontent.com/9159267/69771624-b4fdb780-114a-11ea-8d84-4842aa8f22e2.png)

I replaced the apply calls with regular apply() call and the memory doesn't seem to accumulate. Here is a test script:

```
import numpy as np
import pandarallel
import pandas as pd

NUM_WORKERS = 10
USE_PARALLEL = True
test_data_df = pd.DataFrame(np.random.rand(100, 100))

def random_func(row):
    return row ** 2
if USE_PARALLEL:
    for i in range(10):
        pandarallel.initialize(nb_workers=NUM_WORKERS)
        test_data_df.parallel_apply(random_func)
        print(i)
else:
    for i in range(10):
        test_data_df.apply(random_func)
        print(i)
```

I looked at the issue board and most relevant issue I found was: https://github.com/nalepae/pandarallel/issues/33 

However I am using pandarallel 1.4.1 and it no longer has plasma backend. 

Is there a way to flush the memory after every parallel_apply call?

**Comment by nalepae** [comment]
Thanks for your very well written bug report.
I also reproduce the issue in my tests.

**Comment by nalepae** [comment]
Fixed in `pandarallel v1.4.2`.

**Comment by kkawabat** [comment]
Hi @nalepae I am not sure if this is part of the same issue or something else but I have a dataframe with 70k+ rows and parallel_apply(function) would freeze after processing half the files. I think the problem is with pandarallel since same things have happened with different functions and it was able to process fine with regular apply() call. 

I don't think it's a memory problem since only 11.3 out of 31.3 Gb of ram is being used and I can stop the script just fine. 

I believe it get's stuck in this while loop inside pool.py's[ handle_workers()](https://github.com/python/cpython/blob/894331838b256412c95d54051ec46a1cb96f52e7/Lib/multiprocessing/pool.py#L512) call:

```
            while thread._state == RUN or (pool._cache and thread._state != TERMINATE):
            pool._maintain_pool()
            time.sleep(0.1)
```

I am not sure what the best way to debug this is. Is there anything else I can provide that might help.
